sacc2/sacc.py

```diff
@@ -0,0 +1,871 @@
+
+        Returns
+        -------
+        values: list of lists
+            For each input tag, a corresponding list of the value of that tag for given
+            selection, in the order the matching data points were added.
+
+
+        """
+        indices = set(indices)
+        values = [[d.get_tag(tag) for i, d in enumerate(self.data) if i in indices]
+                  for tag in tags]
+        return values
+
+    def get_tags(self, tags, data_type=None, tracers=None, **select):
+        """
+        Get the value of a one or more named tags for (a subset of) the data.
+
+        Parameters
+        ----------
+
+        tags: list of str
+            Tags to look up on the selected data
+
+        data_type: str
+            Select only data points which are of this data type.
+            If None (the default) then match any data types
+
+        tracers: tuple
+            Select only data points which match this tracer combination.
+            If None (the default) then match any tracer combinations.
+
+        **select:
+            Select only data points with tag names and values matching
+            all values provided in this kwargs option.
+            You can also use the syntax name__lt=value or
+            name__gt=value in the selection to select points
+            less or greater than a threshold
+
+        Returns
+        -------
+        values: list of lists
+            For each input tag, a corresponding list of the value of that tag for given
+            selection, in the order the matching data points were added.
+
+
+        """
+        indices = self.indices(data_type=data_type, tracers=tracers, **select)
+        return self._get_tags_by_index(tags, indices)
+
+    def get_tag(self, tag, data_type=None, tracers=None, **select):
+        """
+        Get the value of a one tag for (a subset of) the data.
+
+        Parameters
+        ----------
+
+        tag: str
+            Tag to look up on the selected data
+
+        data_type: str
+            Select only data points which are of this data type.
+            If None (the default) then match any data types
+
+        tracers: tuple
+            Select only data points which match this tracer combination.
+            If None (the default) then match any tracer combinations.
+
+        **select:
+            Select only data points with tag names and values matching
+            all values provided in this kwargs option.
+            You can also use the syntax name__lt=value or
+            name__gt=value in the selection to select points
+            less or greater than a threshold
+
+        Returns
+        -------
+        values: list
+            A list of the value of the tag for given selection,
+            in the order the matching data points were added.
+
+
+        """
+        return self.get_tags([tag], data_type=data_type, tracers=tracers, **select)[0]
+
+    def get_data_points(self, data_type=None, tracers=None, **select):
```

Comment:
I was thinking of cases where you have more rich tags than we have for 3x2pt, to avoid the need to do repeated searches in get_tag.  But that might be a bit farfetched.  Could maybe delete this or make it internal only?

Reply:
We can leave it there in case it's useful (and delete later if it turns out not to be). Just wanted to understand the logic.